internal/backend/runtime/omni/controllers/omni/machine_status.go

```diff
@@ -417,7 +417,8 @@ func (ctrl *MachineStatusController) handleNotification(ctx context.Context, r c
 			spec.PlatformMetadata = event.PlatformMetadata
 		}
 
-		if event.TalosMachineStatus != nil {
+		if event.TalosMachineStatus != nil && (spec.TalosMachineStatus == nil ||
+			spec.TalosMachineStatus.Version < event.TalosMachineStatus.Version) {
```

Comment:
here we compare versions, but on reboot they reset to '1', so not sure how it would succeed until it reaches the version it was before?

Reply:
I think my question here is still valid... machine was running, version of `TalosMachineStatus` went through e.g. 1..2..3.. ... 128, then it reboots and version starts counting from 1 once again.

Reply:
yeah. Then I don't think we have a way to limit the updates frequency.
Unless we rely on the updated at of the `MachineStatus`, which is also unreliable.

Reply:
unless we also count reboot count of the node and take that into equation, but that's too complicated...